src/ProgramBlockEditor.js

```diff
@@ -247,10 +277,59 @@ class ProgramBlockEditor extends React.Component<ProgramBlockEditorProps, Progra
         this.insertSelectedCommandIntoProgram(stepNumber);
     };
 
+    // TODO: Discuss removing this once we have a good way to test drag and drop.
     /* istanbul ignore next */
-    handleDropCommand = (stepNumber: number) => {
-        this.insertSelectedCommandIntoProgram(stepNumber);
-    };
+    handleDragCommandOverProgramArea = (event: DragEvent) => {
+        if (!this.props.editingDisabled) {
+            event.preventDefault();
+
+            // Only attempt to recalculate the closest node every 100ms.
+            const timeStamp = Date.now();
+            if (timeStamp - this.lastCalculatedClosestAddNode > 100) {
+                const closestAddNodeIndex = this.findAddNodeClosestToEvent(event);
+                this.lastCalculatedClosestAddNode = timeStamp;
+
+                this.setState({
+                    closestAddNodeIndex: closestAddNodeIndex
+                });
+            }
+        }
+    }
+
+    // TODO: Discuss removing this once we have a good way to test drag and drop.
+    /* istanbul ignore next */
+    handleDragLeaveOnProgramArea = (event: DragEvent) => {
+        if (!this.props.editingDisabled) {
+            // Ignore anything that we "contain", except for our upstairs and lefthand neighbours.
+            if (!event.relatedTarget ||
+                // $FlowFixMe: Flow doesn't recognise the relatedTarget property.
+                event.relatedTarget.className === "ProgramBlockEditor__header" ||
+                event.relatedTarget.className === "ProgramBlockEditor__container" ||
+                !this.programBlockEditorRef.current.contains(event.relatedTarget)
```

Comment:
As I mentioned in a previous comment, you are testing here for containment inside `this.programBlockEditorRef`. This is not the element to which you have attached the drag event handlers. If you change to checking for containment inside `this.programSequenceContainerRef` instead, you can remove the special casing for `"ProgramBlockEditor__header"` and `"ProgramBlockEditor__container"`.

Reply:
Sorry, I misread your last comment as explaining why I could continue doing something, I didn't catch the difference in the names.  It's fixed now.